intelmq/bin/intelmqctl.py

```diff
@@ -346,7 +349,237 @@ def __status_process(self, pid, module, bot_id):
+        module = self.__runtime_configuration[bot_id]["module"]
+        cmdargs = [module, bot_id]
+
+        self._get_supervisor().twiddler.addProgramToGroup(self.SUPERVISOR_GROUP, bot_id, {
+            "command": " ".join(cmdargs),
+            "stopsignal": "INT",
+        })
+
+    def _remove_bot(self, bot_id: str):
+        self._get_supervisor().twiddler.removeProcessFromGroup(self.SUPERVISOR_GROUP, bot_id)
+
+    def _get_process_state(self, bot_id: str):
+        try:
+            return self._get_supervisor().supervisor.getProcessInfo(self._process_name(bot_id))["state"]
+        except xmlrpc.client.Fault as e:
+            if e.faultCode == self.RpcFaults.BAD_NAME:
+                return None
+            raise
+
+    def _get_supervisor(self) -> xmlrpc.client.ServerProxy:
+        class UnixStreamHTTPConnection(http.client.HTTPConnection):
+            def connect(self):
+                self.sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+                self.sock.connect(self.host)
+
+        class UnixStreamTransport(xmlrpc.client.Transport, object):
+            def __init__(self, socket_path):
+                self.socket_path = socket_path
+                super(UnixStreamTransport, self).__init__()
+
+            def make_connection(self, host):
+                return UnixStreamHTTPConnection(self.socket_path)
+
+        if not self.__supervisor_xmlrpc:
+            socket_path = os.environ.get("SUPERVISOR_SOCKET", self.DEFAULT_SOCKET_PATH)
+
+            if not os.path.exists(socket_path):
+                raise Exception("Socket '{}' does not exists. Is supervisor running?".format(socket_path))
+
+            if not os.access(socket_path, os.W_OK):
+                raise Exception("Socket '{}' is not accessible. Do you have write permission?".format(socket_path))
+
+            self.__supervisor_xmlrpc = xmlrpc.client.ServerProxy(
+                "http://none",
+                transport=UnixStreamTransport(socket_path)
+            )
+
+            self.__logger.debug("Connected to supervisor '{}' (API version {})".format(
+                self.__supervisor_xmlrpc.supervisor.getIdentification(),
+                self.__supervisor_xmlrpc.supervisor.getAPIVersion()
+            ))
+
+            supervisor_state = self.__supervisor_xmlrpc.supervisor.getState()
+
+            if supervisor_state["statename"] != "RUNNING":
+                raise Exception("Unexpected supervisor state {}".format(supervisor_state["statename"]))
+
+            try:
+                self.__supervisor_xmlrpc.twiddler.getAPIVersion()
+            except xmlrpc.client.Fault:
+                raise Exception("Twiddler is not supported. Is Twiddler for supervisor installed and enabled?")
+
+            if self.SUPERVISOR_GROUP not in self.__supervisor_xmlrpc.twiddler.getGroupNames():
+                raise Exception("Supervisor does not contains process group {}. It must be created manually.")
```

Comment:
Looks like you forgot to format that string

Reply:
Yes, fixed.